### ISSUE TITLE
PairwiseDistancePotential returns NaN autograd gradients for clash-only bounds

### ISSUE THREAD

**Issue by dajiaohuang** [bytedance/Protenix#336]
## Description

`PairwiseDistancePotential` assigns `+inf` as the upper bound for clash-only pairs. `_flat_bottom_parabolic` squares `value - upper` before masking the inactive upper-bound branch with `torch.where`. The forward energy is finite, but autograd differentiates through the `(-inf) ** 2` expression and returns NaN gradients.

This affects the denoiser-path TFG guidance path when `rho` is nonzero, because that path differentiates `Potential.energy(...)` with autograd. The default `rho = 0` path is not affected.

## Reproduction

Tested on current `main` at commit `4c355be4553512f72453ecbfb65e69f4c35d1413` with Python 3.13.13 and PyTorch 2.12.0.dev20260408+cu128:

```python
import torch
from protenix.tfg.potentials import _flat_bottom_parabolic

value = torch.tensor([0.5], requires_grad=True)
energy, analytic_grad = _flat_bottom_parabolic(
    value,
    torch.ones_like(value),
    torch.tensor([2.0]),
    torch.tensor([float("inf")]),
)
autograd_grad, = torch.autograd.grad(energy.sum(), value)

print(energy)         # tensor([1.1250])
print(analytic_grad)  # tensor([-1.5000])
print(autograd_grad)  # tensor([nan])
```

The same result occurs through the real `PairwiseDistancePotential.energy(...)` caller for a two-atom nonbonded clash pair: `_get_distance_bounds(...)` returns a finite lower bound and `upper = inf`; `energy_and_grad(...)` returns a finite analytic coordinate gradient, while `torch.autograd.grad(potential.energy(...), coords)` returns NaN for every coordinate component.

## Expected behavior

Autograd through `PairwiseDistancePotential.energy(...)` should remain finite and match `energy_and_grad(...)` for clash-only bounds.

## Suggested fix

Clamp each violation to zero before squaring it, matching the helper's documented formula (`max(0, violation)`). This avoids constructing an infinite inactive branch while preserving finite-bound energies and gradients. A regression test can compare autograd with `energy_and_grad(...)` through a real clash-only `PairwiseDistancePotential` call and cover finite bounds separately.
